### DISCUSSION TITLE
Is there a way to configure AutoGen to use only custom APIs?

### DISCUSSION THREAD

**Discussion by JoseClua** [microsoft/autogen#2485] [Q&A] [answered]
We have our own set of APIs and we are experimenting the idea of having AI Agents autonomously using these APIs but we would like to force them to not use anything else. In other words, evreything should go through our APIs.
Is there a code sample of:

1. how to assign these APIs to the agents?
2. how to make the agents use the query parameters, uri paramenters?
3. Can they understand swagger?

Thanks

**Comment by ekzhu** [comment, marked as answer]
You can wrap you APIs through tools

See: https://microsoft.github.io/autogen/docs/tutorial/tool-use

**Reply by JoseClua** [reply]
That is exactly what I was looking for. Thanks!
